src/LiveComponent/src/Metadata/LiveComponentMetadataFactory.php

```diff
@@ -115,10 +119,16 @@ public function createLivePropMetadata(string $className, string $propertyName,
                 $collectionValueType
             );
         } else {
-            $type = $this->propertyTypeExtractor->getType($className, $property->getName());
+            $infoType = $this->propertyTypeExtractor->getType($className, $property->getName());
+
+            $collectionValueType = $infoType instanceof CollectionType ? $infoType->getCollectionValueType() : null;
 
-            if ($type instanceof UnionType && !$type instanceof NullableType || $type instanceof IntersectionType) {
-                throw new \LogicException(\sprintf('Union or intersection types are not supported for LiveProps. You may want to change the type of property "%s" in "%s".', $propertyName, $className));
+            if (null !== $collectionValueType && null !== $infoType) {
```

Comment:
```suggestion
            if ($infoType instance of CollectionType) {
```
Seems to be enough no?

Reply:
I think yes, it would ease the readability too

Reply:
Applied, thanks